wrappers/services.go

```diff
@@ -61,18 +61,30 @@ func generateSnapServiceFile(app *snap.AppInfo) ([]byte, error) {
 func stopService(sysd systemd.Systemd, app *snap.AppInfo, inter interacter) error {
 	serviceName := app.ServiceName()
 	tout := serviceStopTimeout(app)
-	if err := sysd.Stop(serviceName, tout); err != nil {
-		if !systemd.IsTimeout(err) {
+
+	stop := func(serviceName string) error {
+		if err := sysd.Stop(serviceName, tout); err != nil {
+			if !systemd.IsTimeout(err) {
+				return err
+			}
+			inter.Notify(fmt.Sprintf("%s refused to stop, killing.", serviceName))
```

Comment:
I'm not sure this logic makes sense for sockets. There's nothing to term or kill there. I think we can preserve the original logic here without the closure, and use simpler stops for the sockets.